tests/test_tensorflow_sparse_data_api.py

```diff
@@ -0,0 +1,243 @@
+    # TODO: TF https://github.com/tensorflow/tensorflow/issues/47532
+    # TODO: TF https://github.com/tensorflow/tensorflow/issues/47931
+    model.compile()
+
+    return model
+
+
+class TestTileDBTensorflowSparseDataAPI(test.TestCase):
+    @testing_utils.run_v2_only
+    def test_tiledb_tf_sparse_data_api_with_with_sparse_data_sparse_label(self):
+        for input_shape in INPUT_SHAPES:
+            with self.subTest():
+                model = create_model(
+                    input_shape=input_shape, num_of_classes=NUM_OF_CLASSES
+                )
+
+                array_uuid = str(uuid.uuid4())
+                tiledb_uri_x = os.path.join(self.get_temp_dir(), "x" + array_uuid)
+                tiledb_uri_y = os.path.join(self.get_temp_dir(), "y" + array_uuid)
+
+                dataset_shape_x = (ROWS, input_shape)
+                dataset_shape_y = (ROWS, (NUM_OF_CLASSES,))
+
+                ingest_in_tiledb_sparse(
+                    uri=tiledb_uri_x,
+                    data=create_sparse_array_one_hot_2d(dataset_shape_x),
+                    batch_size=BATCH_SIZE,
+                )
+                ingest_in_tiledb_sparse(
+                    uri=tiledb_uri_y,
+                    data=create_sparse_array_one_hot_2d(dataset_shape_y),
+                    batch_size=BATCH_SIZE,
+                )
+
+                with tiledb.SparseArray(
+                    tiledb_uri_x, mode="r"
+                ) as x, tiledb.SparseArray(tiledb_uri_y, mode="r") as y:
+                    tiledb_dataset = TensorflowTileDBSparseDataset(
+                        x_array=x, y_array=y, batch_size=BATCH_SIZE
+                    )
+
+                    self.assertIsInstance(tiledb_dataset, tf.data.Dataset)
+                    model.fit(tiledb_dataset, verbose=0, epochs=2)
+
+    @testing_utils.run_v2_only
+    def test_tiledb_tf_sparse_data_api_with_sparse_data_dense_label(self):
+        for input_shape in INPUT_SHAPES:
+            with self.subTest():
+                model = create_model(
+                    input_shape=input_shape, num_of_classes=NUM_OF_CLASSES
+                )
+
+                array_uuid = str(uuid.uuid4())
+                tiledb_uri_x = os.path.join(self.get_temp_dir(), "x" + array_uuid)
+                tiledb_uri_y = os.path.join(self.get_temp_dir(), "y" + array_uuid)
+
+                dataset_shape_x = (ROWS, input_shape)
+                dataset_shape_y = (ROWS, NUM_OF_CLASSES)
+
+                ingest_in_tiledb_sparse(
+                    uri=tiledb_uri_x,
+                    data=create_sparse_array_one_hot_2d(dataset_shape_x),
+                    batch_size=BATCH_SIZE,
+                )
+                ingest_in_tiledb(
+                    uri=tiledb_uri_y,
+                    data=np.random.rand(*dataset_shape_y),
+                    batch_size=BATCH_SIZE,
+                )
+
+                with tiledb.SparseArray(tiledb_uri_x, mode="r") as x, tiledb.DenseArray(
```

Comment:
Same comment about `tiledb.open()`

Reply:
I just pre-emptively used it as directed in the dense case for possible leaks [in the dense case PR](https://github.com/TileDB-Inc/TileDB-ML/pull/25#discussion_r621353086)

Reply:
It's not `tiledb.open()` that's leaking, it's the lack of `with` expression. Here you'll keep using `with`:
```
with tiledb.open(tiledb_uri_x) as x, tiledb.open(tiledb_uri_y) as y:
...
```